openassessment/fileupload/api.py

```diff
@@ -53,6 +53,50 @@ def get_student_file_key(student_item_dict, index=0):
     return key_template.format(index=index, **student_item_dict)
 
 
+def can_delete_file(teams_enabled, key, team_id, shared_file=None):
+    """
+    A user is allowed to delete any file they own if this is not a team-enabled response.
+    If the response is team-enabled, a user, who is a member of a team,
+    is allowed to delete a file they own as long as they are still
+    a member of the team with which the file has been shared.
+
+    params:
+      teams_enabled (boolean): Indicates if teams are enabled for an ORA block.
+      key (string): The key of the file to check if we can delete.
+      team_id (string): The id of the team of the user who may be able to delete the file.
+      shared_file (SharedFileUpload): Optional. A SharedFileUpload object corresponding to the given
+      key.  It's useful to pass this in if you've already fetched all of the SharedFileUpload records
+      for a given item/team.
+
+    raises:
+        SharedFileUpload.DoesNotExist If teams are enabled, a team_id is provided,
+        and no SharedFileUpload corresponding to the file key exists.
+    returns:
+        Boolean indicating if the file with the given key can be deleted by the current user.
+    """
+    if not teams_enabled:
+        return True
+
+    if not shared_file:
+        try:
+            shared_file = SharedFileUpload.by_key(key)
+        except SharedFileUpload.DoesNotExist:
+            # TODO: is it reasonble to say "fine, you can delete it, ugh!"
```

Comment:
Should we at least log a message? Not like that will really do anything ... maybe we should forcibly delete the file in this state xD

Reply:
yeah, added a log statement